platforms/core-runtime/gradle-cli/src/main/java/org/gradle/launcher/cli/BuildActionsFactory.java

```diff
@@ -86,6 +91,10 @@ public void configureCommandLineParser(CommandLineParser parser) {
     @Override
     public Action<? super ExecutionListener> createAction(CommandLineParser parser, ParsedCommandLine commandLine, Parameters parameters) {
         StartParameterInternal startParameter = parameters.getStartParameter();
+        if (commandLine.hasOption(HELP_OPTION) && !startParameter.getTaskRequests().isEmpty()) {
+            TaskExecutionRequest task = startParameter.getTaskRequests().get(0);
+            startParameter.setTaskNames(ImmutableList.of(HELP_TASK, "--task", task.getArgs().get(0)));
```

Comment:
Yep, added a message. It might be hard to see among others:
```
Environment vars:
    JAVA_HOME: /Users/ov7a/.sdkman/candidates/java/11.0.20.1-tem
    GRADLE_HOME: null
    GRADLE_USER_HOME: null
    JAVA_OPTS: null
    GRADLE_OPTS: "-Dorg.gradle.jvmargs='-ea' '-Xms256m' '-Xmx512m' '-XX:MaxMetaspaceSize=512m' '-XX:+HeapDumpOnOutOfMemoryError' '-XX:HeapDumpPath=/Users/ov7a/work/gradle/intTestHomeDir/distributions-full'" -Dorg.gradle.daemon.idletimeout=120000 "-Dorg.gradle.daemon.registry.base=/Users/ov7a/work/gradle/subprojects/diagnostics/build/tmp/teŝt files/HelpTaskInt.Test/g7r34/daemon" -Dorg.gradle.native.dir=/Users/ov7a/work/gradle/intTestHomeDir/distributions-full/native -Dorg.gradle.deprecation.trace=false -Djava.io.tmpdir=/Users/ov7a/work/gradle/subprojects/diagnostics/build/tmp -Dorg.gradle.internal.plugins.portal.url.override=https://plugins.gradle.org/m2 -Dfile.encoding=UTF-8 -Dorg.gradle.classloaderscope.strict=true -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false -Dorg.gradle.kotlin.dsl.skipMetadataVersionCheck=false -ea
Multiple tasks are specified with the `--help` flag. The help is shown for the first task only.
Starting a Gradle Daemon (subsequent builds will be faster)
Executing Gradle on JVM versions 16 and lower has been deprecated. This will fail with an error in Gradle 9.0. Use JVM 17 or greater to execute Gradle. Projects can continue to use older JVM versions via toolchains. Consult the upgrading guide for further information: https://docs.gradle.org/8.11-20240827220000+0000/userguide/upgrading_version_8.html#minimum_daemon_jvm_version
```
but I think it's good enough